### PR TITLE
feat: IAM Auth for RDS

### PR DESCRIPTION
**Pull request by jvmakine**
Have not been yet fully test e2e, but at least this creates the user with right auth mechanism. 

Merging this in will help testing, as I won't need to keep manual docker images in sync.

### CODE REVIEW

**Review by stuartwdouglas — APPROVED** [review]
I don't really like how these dependencies are leaking into user deployments, but that is not something we can solve now.

File: internal/dsn/dsn.go [hunk 1]
```diff
@@ -45,11 +48,25 @@ func MySQLDSN(dbName string, options ...Option) string {
...
+	case *schema.DSNDatabaseConnector:
+		return c.DSN, nil
+	case *schema.AWSIAMAuthDatabaseConnector:
+		cfg, err := config.LoadDefaultConfig(ctx)
+		if err != nil {
+			return "", fmt.Errorf("configuration error: %w", err)
+		}
+
+		authenticationToken, err := auth.BuildAuthToken(
+			// TODO: proper region
```
Comment (alecthomas) [inline internal/dsn/dsn.go:61]: !!

File: internal/schema/database.go [hunk 2]
```diff
@@ -139,7 +139,39 @@ func (d *DSNDatabaseConnector) ToProto() protoreflect.ProtoMessage {
...
 func (d *DSNDatabaseConnector) schemaChildren() []Node { return nil }
-func (d *DSNDatabaseConnector) schemaSymbol()          {}
+
+//protobuf:2
+type AWSIAMAuthDatabaseConnector struct {
+	Pos Position `parser:"" protobuf:"1,optional"`
+
+	Username string `parser:"" protobuf:"2"`
+	Endpoint string `parser:"" protobuf:"3"`
+	Database string `parser:"" protobuf:"4"`
```
Comment (alecthomas) [inline internal/schema/database.go:149]: We don't want this to be parseable?

**Review by alecthomas — APPROVED** [review]

File: internal/dsn/dsn.go (hunk 1 above)
Comment (jvmakine) [reply internal/dsn/dsn.go:61]: Will be done very soon!

File: internal/schema/database.go (hunk 2 above)
Comment (jvmakine) [reply internal/schema/database.go:149]: Hmm, that is just cargo culting from the existing runtime, and it seems to work. Not totally sure what this does. Should these be parseable?

**Pull request merged**